### PR TITLE
Jul21 invitation

### PR DESCRIPTION
**Pull request by talmawash**
## Description

Now a user can join a list using a code. This is facilitated with a custom add list view that gives the user an option to either join or create a new list. Previously, the MainViewController handled list addition, but with this separate view, I had to create a delegate that relays information back to the main view controller. 

This PR makes use of the pod ProgressHUD which is a nice addition

## Milestones
The app became usable without manual database modification at this stage. Additionally, going through this method of showing a custom view should be helpful if I decide to use it for other features.

## Test Plan
Kapture 2022-07-21 at 15 53 40

### CODE REVIEW

File: Backend/lists.js [hunk 1]
```diff
@@ -25,15 +24,49 @@ Parse.Cloud.define("newList", async (request) => {
...
   requireUser: true
 });
+
+Parse.Cloud.define("joinList", async (request) => {
+  const invitations = await new Parse.Query("SharecartInvitation").equalTo("objectId", request.params.code).find({ useMasterKey: true });
+  if (invitations.length == 1) {
+      const invitation = invitations[0];
+      
+      if (invitation.used) {
+          throw("Invitation already used!");
```
Comment (franklee77) [inline Backend/lists.js:45]: Do you want to throw here? That'll crash the app. It'll be better to return an error code. Ideally we'd define an enum with the possible responses from joinList call.

File: Backend/lists.js [hunk 2]
```diff
@@ -25,15 +24,49 @@ Parse.Cloud.define("newList", async (request) => {
...
+      }
+      
+      const hourDifference = (Date.now() - invitation.createdAt) / (1000 * 60 * 60)
+      if (hourDifference >= 24) {
+          throw("Invitation exipred!");
```
Comment (franklee77) [inline Backend/lists.js:50]:
```suggestion
          throw("Invitation expired!");
```

Comment (franklee77) [reply Backend/lists.js:50]: Plus same comment regarding this use of throw.

File: Backend/lists.js (hunk 1 above)
Comment (talmawash) [reply Backend/lists.js:45]: this sends the error message back to the app, not sure if it has any consequence on the server side

File: Backend/lists.js [hunk 3]
```diff
@@ -25,15 +24,49 @@ Parse.Cloud.define("newList", async (request) => {
...
+Parse.Cloud.define("joinList", async (request) => {
+  const invitations = await new Parse.Query("SharecartInvitation").equalTo("objectId", request.params.code).find({ useMasterKey: true });
+  if (invitations.length == 1) {
+      const invitation = invitations[0];
+      
+      if (invitation.used) {
+          throw("Invitation already used!");
+      }
+      
+      const hourDifference = (Date.now() - invitation.createdAt) / (1000 * 60 * 60)
```
Comment (franklee77) [inline Backend/lists.js:48]: Let's make this a constant up top: MILLISECOND_PER_HOUR

**Review by franklee77 — APPROVED** [review]

**Pull request merged**